picard/ui/profileeditor.py

```diff
@@ -0,0 +1,454 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2021 Bob Swift
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+from copy import deepcopy
+import uuid
+
+from PyQt5 import (
+    QtCore,
+    QtWidgets,
+)
+
+from picard.config import (
+    IntOption,
+    ListOption,
+    Option,
+    get_config,
+)
+from picard.const import PICARD_URLS
+from picard.profile import UserProfileGroups
+
+from picard.ui import (
+    PicardDialog,
+    SingletonDialog,
+)
+from picard.ui.moveable_list_view import MoveableListView
+from picard.ui.options import OptionsPage
+from picard.ui.ui_profileeditor import Ui_ProfileEditorPage
+from picard.ui.widgets.profilelistwidget import ProfileListWidgetItem
+
+
+class ProfileEditorDialog(SingletonDialog, PicardDialog):
+    """User Profile Editor Page
+    """
+    TITLE = N_("User profile editor")
+    STYLESHEET_ERROR = OptionsPage.STYLESHEET_ERROR
+
+    help_url = PICARD_URLS["doc_profile_edit"]
+
+    PROFILES_KEY = "user_profiles"
+    SETTINGS_KEY = "user_profile_settings"
+    POSITION_KEY = "last_selected_profile_pos"
+
+    options = [
+        ListOption.add_if_missing("profiles", PROFILES_KEY, []),
+        Option.add_if_missing("profiles", SETTINGS_KEY, {}),
```

Comment:
Are those really needed (as they are declared in `SettingConfigSection.__init__()`) ?